### PR TITLE
Dashboard frontend assets

### PR DESCRIPTION
**Pull request by beckyvb**
Refactor scss.
Add toaster scss.
Add many docs to the scss/styleguide.
Edit styleguide asset compilation and workflow.
Tweak javascript.

## Notes on this PR:

* Anything in the `template` directory other than the kss* assets, image assets and index.html is autogenerated from the dashboard application assets. The template is used to compile the styleguide.
* `js/dashboard.js` is autogenerated from the files in `js/src/`
* `admin.css` is generated from the scss files

Given the above, what needs to be reviewed are changes to `main.js`, `modal.dashboard.js`, `jquery.fluidfixed.js`, `Gruntfile.js` (for workflow changes), the scss files (stay encouraged!), markdown docs, and the php files.

## Before You Merge!

I've made a PR off this branch for the compiled styleguide. Otherwise, the PR was large to review properly in github. Merge this PR into this one before merging altogether: https://github.com/vanilla/vanilla/pull/4635

To view the styleguide, you can checkout the `fix/compiled-styleguide` branch and look at /styleguide/index.html in your browser.

### CODE REVIEW

File: applications/dashboard/js/jquery.fluidfixed.js [hunk 1]
```diff
@@ -63,6 +64,8 @@
                     // We're going to extend past the page bottom.
                     $(element).css('margin-top', vars.maxMarginDiff);
                 }
+            } else {
```
Comment (beckyvb) [inline applications/dashboard/js/jquery.fluidfixed.js]: Fixes issue on a page when the nav is longer than the page height when expanded but shorter when collapsed. Collapsing the nav makes it fixed, but expanding it again doesn't un-fix it, so the nav gets cut off.

File: applications/dashboard/Gruntfile.js [hunk 2]
```diff
@@ -206,21 +199,13 @@ module.exports = function (grunt) {
...
-    },
-
     concat: {
       dist: {
         src: ([]).concat([
           'js/src/lithe.js',
           'js/src/lithe.drawer.js',
           'js/src/modal.dashboard.js',
-          'js/src/main.js'
+          'js/src/main.js',
```
Comment (DaazKu) [inline applications/dashboard/Gruntfile.js]: That quote should not be there.

File: applications/dashboard/scss/extensions/README.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# SCSS Extensions
+
+These are adapter stylesheets that map old ways of doing things to new ways. Optimally, we want this directory to be
+empty, but there's some work to be done before we get there. Each file is annotated with it's raison d'être and/or what
```
Comment (DaazKu) [inline applications/dashboard/scss/extensions/README.md]: its raison d'être

**Comment by DaazKu** [comment]
`.Hidden` does not work on inputs anymore.

screen shot 2016-10-13 at 11 34 02 am

**Comment by DaazKu** [comment]
Profile pictures do not take the container size anymore.

screen shot 2016-10-13 at 11 45 25 am

**Comment by DaazKu** [comment]
Radios are misaligned

screen shot 2016-10-13 at 11 54 49 am

**Pull request merged**